test/e2e/framework/framework.go

```diff
@@ -186,7 +253,9 @@ func (f *Framework) setupNamespace(ctx context.Context) {
 	}, metav1.CreateOptions{})
 	o.Expect(err).NotTo(o.HaveOccurred())
 
-	// Grant it edit permission in this namespace.
+	f.userSA = userSA
+
+	// Grant userSA edit permission in parent namespace.
```

Comment:
nit: lets call these relations consistently, here you use "parent" in other "primary".

Reply:
done